extensions/mssql/src/dashboard/bookExtensions.ts

```diff
@@ -86,22 +86,25 @@ export interface BookContributionProvider {
 
 class AzdataExtensionBookContributionProvider extends Disposable implements BookContributionProvider {
 	private _contributions?: BookContribution[];
+	private _contributionCommands: vscode.Disposable[] = [];
 
 	public constructor(
 		public readonly extensionPath: string,
 	) {
 		super();
 
-		vscode.extensions.onDidChange(() => {
+		vscode.extensions.onDidChange(async () => {
 			const currentContributions = this.getCurrentContributions();
 			const existingContributions = this._contributions || undefined;
 			if (!arrays.equals(existingContributions, currentContributions, BookContributions.equal)) {
+				await this.unregisterCommands();
 				this._contributions = currentContributions;
+				await this.registerCommands();
```

Comment:
What catches and bubbles up the error if this throws?

Reply:
Extension host will just display it in the console log. There's not really anything useful we can have the customer do and this isn't something that a user should ever run in to so don't really see the value in doing anything else here.